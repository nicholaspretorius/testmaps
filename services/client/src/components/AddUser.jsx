import React from "react";
import PropTypes from "prop-types";

const AddUser = props => {
  const { addUser, email, onHandleChange } = props;

  return (
    <form onSubmit={e => addUser(e)}>
      <div className="field">
        <label htmlFor="input-email" className="label is-large">
          Email
        </label>
        <input
          type="email"
          name="email"
          id="input-email"
          className="input is-large"
          placeholder="Enter your email address"
          value={email}
          onChange={onHandleChange}
          required
        />
      </div>
      <input
        type="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
      />
    </form>
  );
};

AddUser.propTypes = {
  email: PropTypes.string.isRequired,
  onHandleChange: PropTypes.func.isRequired,
  addUser: PropTypes.func.isRequired
};

export default AddUser;
