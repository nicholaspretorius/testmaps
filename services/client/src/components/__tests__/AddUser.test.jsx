import React from "react";
import { render, cleanup } from "@testing-library/react";

import AddUser from "./../AddUser";

afterEach(cleanup);

it("renders with default props", () => {
  const { getByLabelText, getByText } = render(<AddUser email="" handleChange={() => true} />);

  const emailInput = getByLabelText("Email");
  expect(emailInput).toHaveAttribute("type", "email");
  expect(emailInput).toHaveAttribute("required");
  expect(emailInput).not.toHaveValue();

  const buttonInput = getByText("Submit");
  expect(buttonInput).toHaveValue("Submit");
});

it("renders", () => {
  const { asFragment } = render(<AddUser email="" handleChange={() => true} />);
  expect(asFragment()).toMatchSnapshot();
});
