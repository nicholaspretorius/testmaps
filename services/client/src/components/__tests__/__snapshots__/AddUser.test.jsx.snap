// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <form>
    <div
      class="field"
    >
      <label
        class="label is-large"
        for="input-email"
      >
        Email
      </label>
      <input
        class="input is-large"
        id="input-email"
        name="email"
        placeholder="Enter your email address"
        required=""
        type="email"
        value=""
      />
    </div>
    <input
      class="button is-primary is-large is-fullwidth"
      type="submit"
      value="Submit"
    />
  </form>
</DocumentFragment>
`;
